packages/datadog-instrumentations/src/http/client.js

```diff
@@ -0,0 +1,157 @@
+'use strict'
+
+const url = require('url')
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('../helpers/instrument')
+const shimmer = require('../../../datadog-shimmer')
+
+const log = require('../../../dd-trace/src/log')
+
+const startClientCh = channel('apm:http:client:request:start')
+const asyncEndClientCh = channel('apm:http:client:request:async-end')
+const endClientCh = channel('apm:http:client:request:end')
+const errorClientCh = channel('apm:http:client:request:error')
+
+addHook({ name: 'https' }, http => {
```

Comment:
The callback passed into both of these is exactly the same. Maybe it should just be single function passed to both, like:

```js
function hookFn (http) {
  patch.call(this, http, 'request')
  patch.call(this, http, 'get')
  
  return http
}
```

In addition, there doesn't appear to be any reason to use `call` here with `this`, since it's currently an arrow function, so `this` is just the enclosing `this`, which is `global`. Also, the `patch` function doesn't use `this` at all.